Make sure there is a maintainer for every testcase

emanchado opened the pull request:
Now that we can specify maintainers at the test case level, we want to
allow specifying _only_ at that level and not at the suite level. That
failed before, so fix that and make sure that every test case has a
maintainer, either from the case itself, or from its containing suite.

spbnick commented on kpet/data.py:
```diff
@@ -370,6 +370,8 @@ def __init__(self, data):
             self.name = self.description
         self.validate_case_ids()
+        if self.maintainers is None:
+            self.maintainers = []
```
Can you make the above two changes a separate commit?

spbnick commented on kpet/data.py:
```diff
@@ -563,6 +565,14 @@ def get_sets_from_regex_list(regex_list):
+                                  "test, either at suite level or at case "
+                                  "level; in suite: {}\ncase: {}".
+                                  format(suite.name, case.name))
+
```
This is an inappropriate place for such a check, if only because it has nothing to do with sets. Instead please put this into `kpet.data.Suite.__init__()`

spbnick reviewed (changes requested):
Apart from the inline comments, could you please either rework or even completely drop the test? It's too complex for such a small piece of functionality. I think the test could be done entirely in code, with only minimal template handling only the maintainers.

emanchado committed:
Streamline test suite maintainer validation

Don't require the list to have at least one maintainer (it's optional,
after all) and make sure it's always a list, and never None. This
mimics how the test case maintainers work.

emanchado committed:
Make sure there is a maintainer for every testcase

Now that we can specify maintainers at the test case level, we want to
allow specifying _only_ at that level and not at the suite level. That
failed before, so fix that and make sure that every test case has a
maintainer, either from the case itself, or from its containing suite.

spbnick commented on kpet/data.py:
```diff
@@ -457,6 +457,20 @@ def validate_host_type_regex(self):
+                                  "test, either at suite level or at case "
+                                  "level; in suite: {}\ncase: {}".
+                                  format(suite.name, case.name))
+
```
Could you please put this into `kpet.data.Suite.__init__()` instead? It only concerns the test suites separately, we don't need the context of the whole database to check cases have maintainers, only the suite context.
